File: examples/cpp/client_example.cc
#include "client.h"

#include <iostream>

int main() {
  Client::Config config;
  config.transport.insecure = true;

  Client client("127.0.0.1:50051", "orders.primary", config);
  const auto ranges = client.AllocateTimestamps(1);

  std::cout << "tso=" << ranges.front().start_tso() << std::endl;
  return 0;
}

File: examples/rust/client_example.rs
use chronos::{Client, ClientConfig, ClientTransportConfig};

type AppResult<T> = Result<T, Box<dyn std::error::Error>>;

#[tokio::main]
async fn main() -> AppResult<()> {
    let config = ClientConfig::new("orders.primary")
        .with_transport(ClientTransportConfig::default().with_insecure(true));
    let client = Client::connect_with_config("127.0.0.1:50051", config).await?;
    let ranges = client.allocate_timestamps(1).await?;

    println!("tso={}", ranges[0].start_tso);
    Ok(())
}
